modules/README.md

```diff
@@ -115,35 +119,235 @@ Kudos to @mfw78
+
+```mermaid
+sequenceDiagram
+	participant Account
+    participant SafeProtocolManager
+	participant RegistryContract
+
+    Account->>SafeProtocolManager: Enable SignatureValidatorManager as a function handler for function selector of `isSignatureValid(bytes32,bytes)` for the account
+    SafeProtocolManager->>RegistryContract: Fetch from registry information about SignatureValidatorManager
+    RegistryContract-->>SafeProtocolManager: Return result
+    SafeProtocolManager->>SafeProtocolManager: Check if SignatureValidatorManager is listed and not flagged
+    SafeProtocolManager->>SafeProtocolManager: Check if SignatureValidatorManager implements ISafeProtocolFunctionHandler interface
+    SafeProtocolManager-->>Account: Return result
+```
 
+### Signature validator interface
 
-- EIP-712 based Signature Validator
+A signature validator must implement the following interface.
 
 ```solidity
-interface ISafeProtocol712SignatureValidator {
+interface ISafeProtocolSignatureValidator {
     /**
      * @param safe The Safe that has delegated the signature verification
      * @param sender The address that originally called the Safe's `isValidSignature` method
-     * @param _hash The EIP-712 hash whose signature will be verified
+     * @param messageHash The EIP-712 hash whose signature will be verified
      * @param domainSeparator The EIP-712 domainSeparator
-     * @param typeHash The EIP-712 typeHash
-     * @param encodeData The EIP-712 encoded data
-     * @param payload An arbitrary payload that can be used to pass additional data to the validator
+     * @param hashStruct Hash generated from typeHash and encodeData
+     * @param signature The signature to be verified
      * @return magic The magic value that should be returned if the signature is valid (0x1626ba7e)
      */
     function isValidSignature(
         address account,
         address sender,
-        bytes32 _hash,
+        bytes32 messageHash,
         bytes32 domainSeparator,
-        bytes32 typeHash,
-        bytes calldata encodeData,
-        bytes calldata payload
+        bytes32 hashStruct
+        bytes calldata signatures
     ) external view returns (bytes4 magic);
 }
 ```
 
+### Hooks for signature validator
+
+Signature validation is a critical part of smart contract accounts. A flawed implementation of validation logic can lead to approving a transaction with an invalid signature. Signature validator hooks can be used to add safety net to the signature validation process and hold the power to block a signature validator from approving unauthorized actions. Hooks can be enabled for all validators before and after execution of the validation function. This can be done by adding contract implementing `ISignatureValidatorHook` interface to the `SignatureValidatorManager` by the account. The hooks are not specific to a domain and are executed for all signature validations. The hooks function(s) should revert on failed checks.
```

Comment:
Is this hook called for the default signature validation?

Reply:
yes